packages/arb-token-bridge-ui/src/components/TransferPanel/DepositCardSuccess.tsx

```diff
@@ -10,12 +10,26 @@ import {
 } from './DepositCard'
 import { useAppContextDispatch } from '../App/AppContext'
 import { formatBigNumber } from '../../util/NumberUtils'
+import { useNetworksAndSigners } from '../../hooks//useNetworksAndSigners'
+import { useBalance } from 'token-bridge-sdk'
 
 export function DepositCardSuccess({ tx }: { tx: MergedTransaction }) {
   const {
     app: { arbTokenBridge, selectedToken }
   } = useAppState()
   const dispatch = useAppContextDispatch()
+  const {
+    l1: { provider: L1Provider },
+    l2: { provider: L2Provider },
+    isConnectedToArbitrum
+  } = useNetworksAndSigners()
+
+  const {
+    eth: [ethBalance]
+  } = useBalance({
+    provider: isConnectedToArbitrum ? L2Provider : L1Provider,
+    walletAddress: arbTokenBridge.walletAddress
+  })
```

Comment:
In this case we want to show the new balance on L2 after a successful deposit, so we should use the L2 provider always

Reply:
Changed